content/200-concepts/100-components/01-prisma-schema/10-postgresql-extensions.mdx

```diff
@@ -0,0 +1,107 @@
+  previewFeatures = ["postgresExtensions"]
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+```
+
+## How to represent PostgreSQL extensions in your Prisma schema
+
+To represent PostgreSQL extensions in your Prisma schema, add the `extensions` field to the `datasource` block of your `schema.prisma` file with an array of the extensions that you require. For example, the following schema lists the `hstore`, `pg_tgrm` and `postgis` extensions:
+
+```prisma file=schema.prisma
+datasource db {
+  provider = "postgresql"
+  url      = env("TEST_DATABASE_URL")
+  extensions = [hstore(schema: "myHstoreSchema"), pg_tgrm, postgis(version: "2.1")]
+}
+```
+
+Each extension name in the Prisma schema can take the following optional arguments:
+
+- `schema`: the name of the schema in which to install the extension's objects. If this argument is not specified, the current default object creation schema is used.
+- `version`: the version of the extension to install. If this argument is not specified, the value given in the extension's control file is used.
+- `map`: the database name of the extension. If this argument is not specified, the name of the extension in the Prisma schema must match the database name.
+
+In the example above, the `hstore` extension uses the `myHstoreSchema` schema, and the `postgis` extension is installed with version 2.1.
+
+The `map` argument is useful when the PostgreSQL extension you want to install has a name that is not a valid identifier in the Prisma schema. For example, the `uuid-ossp` PostgreSQL extension name is an invalid identifier because it contains a hyphen. In the following example, the extension is mapped to the valid name `uuidOssp` in the Prisma schema:
+
+```prisma file=schema.prisma
+datasource db {
+  provider = "postgresql"
+  url      = env("TEST_DATABASE_URL")
+  extensions = [uuidOssp(map: "uuid-ossp")]
+}
+```
+
+## How to introspect PostgreSQL extensions
+
+To [introspect](/concepts/components/introspection) PostgreSQL extensions currently installed in your database and add relevant extensions to your Prisma schema, run `npx prisma db pull`.
+
+Many PostgreSQL extensions are not relevant to the Prisma schema. For example, some extensions are intended for database administration tasks that do not change the schema. If all these extensions were included, the list of extensions would be very long. To avoid this, Prisma maintains an allowlist of known relevant extensions. The current allowlist is the following:
+
+- `citext`
+- `postgis`
+- `pg_crypto`
+- `uuid-ossp`
+
+Extensions are introspected as follows:
+
+- The first time you introspect, all database extensions that are on the allowlist are added to your Prisma schema
+- When you re-introspect, the behaviour depends on whether the extension is on the allowlist or not.
+  - Extensions on the allowlist:
+    - are **added** if they are in the database but not in the Prisma schema
```

Comment:
Added/kept/removed in the schema file, right, as part of the introspection process? Might be worth calling that our, even though you say it above. Also, consider whether we need to say schema file or schema. I lean towards schema file because that is the object being modified.